lib/concurrent/atomic/atomic_fixnum.rb

```diff
@@ -74,10 +68,7 @@ def value=(value)
     #
     #   @return [Fixnum] the current value after incrementation
     def increment
-      @mutex.lock
-      @value += 1
-    ensure
-      @mutex.unlock
+      synchronize { @value += 1 }
```

Comment:
Yeah I agree @jdantonio. If that is the case, maybe we can name it someting other than atomic_fixnum to make it clear the the behavior doesn't mimic that of Fixnum?